### PR TITLE
Add #timeseries to Sequel::Vertica::Dataset

### PR DESCRIPTION
**Pull request by emschwar**
Sequel::Vertica::Dataset#timeseries takes the following options (all required):
* :alias     => The alias for the time series
* :time_unit => The unit of time of slice computation--
                how long is the timeseries interval, in other words.
* :over      => Specifies partitioning and ordering for the time series.

Example:

DB.from(:foo).select_all.timeseries(alias: :slice_time, time_unit: '1 minute', over: { order: :occurred_at })

At this time, #timeseries does not accept an ORDER BY clause outside of the
:over option, though that is supported by vertica. This is purely driven by
personal need-- I don't need it right now. If you do, feel free to submit a
PR!

### CODE REVIEW

File: lib/sequel/adapters/vertica.rb [hunk 1]
```diff
@@ -126,9 +126,33 @@ def schema_parse_table(table_name, options = {})
...
+      TIMESERIES = ' TIMESERIES '.freeze
+      OVER = ' OVER '.freeze
+      AS = ' AS '.freeze
+
+      Dataset.def_sql_method(self, :select, %w(with select distinct columns from join timeseries where group having compounds order limit lock))
+
+      def timeseries(opts={})
+        fail "timeseries requires :alias" unless opts[:alias]
+        fail "timeseries requires :time_unit" unless opts[:time_unit]
+        fail "timeseries requires an :over clause" unless opts[:over]
```
Comment (camilo) [inline lib/sequel/adapters/vertica.rb]: why fail and not a named exception? this could be an `ArgumentErrror` too

File: lib/sequel/adapters/vertica.rb [hunk 2]
```diff
@@ -126,9 +126,33 @@ def schema_parse_table(table_name, options = {})
...
+
+        clone(timeseries: {
```
Comment (camilo) [inline lib/sequel/adapters/vertica.rb]: why are you calling clone on a newly created object?

File: spec/adapters/vertica_spec.rb [hunk 3]
```diff
@@ -177,13 +177,34 @@ def VERTICA_DB.sqls
   end
 
   describe "Verticas's EXPLAIN and EXPLAIN LOCAL" do
-    specify "should not raise errors" do
+    specify "do not raise errors" do
```
Comment (camilo) [inline spec/adapters/vertica_spec.rb]: don't mind this changes but they should be on a separate commit/PR, there is no reason for a commit adding support for something new, to also have this misc fixes.

File: lib/sequel/adapters/vertica.rb [hunk 4]
```diff
@@ -126,9 +126,33 @@ def schema_parse_table(table_name, options = {})
...
       Database::DatasetClass = self
-      EXPLAIN = 'EXPLAIN '
-      EXPLAIN_LOCAL = 'EXPLAIN LOCAL '
-      QUERY_PLAN = 'QUERY PLAN'
+      EXPLAIN = 'EXPLAIN '.freeze
+      EXPLAIN_LOCAL = 'EXPLAIN LOCAL '.freeze
+      QUERY_PLAN = 'QUERY PLAN'.freeze
+      TIMESERIES = ' TIMESERIES '.freeze
+      OVER = ' OVER '.freeze
+      AS = ' AS '.freeze
```
Comment (camilo) [inline lib/sequel/adapters/vertica.rb]: who is using the new constants?

File: lib/sequel/adapters/vertica.rb (hunk 1 above)
Comment (emschwar) [inline lib/sequel/adapters/vertica.rb]: Gah, sorry, this response is so late. I'll fix this. At the time, rubocop had taken the position that `fail` was preferred to `raise` and I was supporting that. They've since reversed their position, and I see no reason not to use `ArgumentError` here.

File: lib/sequel/adapters/vertica.rb (hunk 2 above)
Comment (emschwar) [inline lib/sequel/adapters/vertica.rb]: That seems to be the pattern I saw with other Sequel adatpers; each method that modifies a dataset returns a new `Dataset` with the options added to it.

File: spec/adapters/vertica_spec.rb (hunk 3 above)
Comment (emschwar) [inline spec/adapters/vertica_spec.rb]: No problem, I will remove this change.

File: lib/sequel/adapters/vertica.rb (hunk 4 above)
Comment (emschwar) [inline lib/sequel/adapters/vertica.rb]: The `#timeseries` method does.

**Comment by camilo** [comment]
travis is not happy, besides that 👍

**Comment by emschwar** [comment]
I don't know how to interpret the travis issue; it seems to say that 1.9.3 can't install?

```An error occurred while installing sequel-vertica (0.3.0), and Bundler cannot
continue.
Make sure that `gem install sequel-vertica -v '0.3.0'` succeeds before bundling.
```

**Comment by emschwar** [comment]
I am suspecting this is related to a Travis bug, based on https://github.com/rubygems/rubygems/issues/1419#issuecomment-166322735

Since 2.0 and 2.1 build correctly, I bet that 1.9.3 would as well, if it had a new enough bundler gem.

**Comment by emschwar** [comment]
See https://github.com/travis-ci/travis-ci/issues/3531#issuecomment-88311203 for a possible workaround

**Comment by camilo** [comment]
hmm I wonder if sequel itself still supports 1.9.

@emschwar can you try and see if the work around makes it work? if not iamma just drop support for 1.9.3 (is 2016 after all) and merge.

**Comment by camilo** [comment]
Actually just rebase master I removed 1.9.3 from travis...

**Comment by camilo** [comment]
I'll merge really if anything breaks in CI I can just revert.
